### ISSUE TITLE
EXP on all data from wollongong

### ISSUE THREAD

**Issue by WeiFoo** [ai-se/HPO#11]
## Settings
- data : apache, duraspace, jboss, jira, moodle, spring, wso2.
- DE: **np = 60** (10 times number of variable),  life = 5,
- Learner: CART and RF
- repeats: 5 times
- tuning goals: F and Precision
## Results
- [resluts for F](https://github.com/ai-se/HPO/issues/10)
- [results for PREC](https://github.com/ai-se/HPO/issues/9)

![1a8ecfc3-340c-4b92-a02b-4b473ffe8ed6](https://cloud.githubusercontent.com/assets/7039841/11605268/5713ecb2-9ac9-11e5-8f19-c4fa1990860d.png)

![22c6556f-5131-45d9-a87d-1a0ec496a369](https://cloud.githubusercontent.com/assets/7039841/11605294/d9f731fc-9ac9-11e5-8e89-ed2245bb294f.png)
## Observation
- For most data sets, tuning does not help a lot(0~5%).
- most naive learners already have pretty good performance before tuning.
- Can we a question: when tuning is helpful? (data distribution?)

**Comment by timm** [comment]
what was the goal here? test the results of the icse paper on new data? 

and was this the same train/test set as icse? i.e. train on past, test on future?

**Comment by WeiFoo** [comment]
the data sets are all from Dr.Dam.

For each project, they have sperate training (A) and testing (B) files. I divided the trianing data into new_training (70%)and new tuning data set(30%) to do tuning. for naive learner, I use A as training data.

**Comment by timm** [comment]
ok... so your mission in life is to be the great tuner. and these data sets are not tuning. even though your prior results (and those of queens) say they should. so what are you going to about it? 
- explore more complex goals (e.g. AUC recall vs precision or AUC false alarm vs recall or what..)?
- explore more elaborate learning? instance selection methods? column selection methods?

over to you. insert your phd here.

**Comment by WeiFoo** [comment]
I will do a quick work tuning AUC pd vs pf on Wollongong data, see if we can get some differnet results.

Then, I probably go for moe relaborate learning stuff.

**Comment by timm** [comment]
you need more results on the value of tuning

your current tuning results are not... divergent. we've only had success on f/prec on a few defect data sets. do you need to do something else? like tune smote on defect data? rahul reports smote can produce large improvements on defect data sets?

**Comment by timm** [comment]
hey, i note that smote does not help much when the version0 results have high f,prec values.

maybe smote is good for helping BAD learners but is not so useful when the off-the-shelf learner is GOOD.

**Comment by WeiFoo** [comment]
@timm tune wollongong data on AUC is done, but the results are same as prec/F.

Probably, we can get the conclusion: Tuning can't improve much when baseline results are good enough.
[TUNE AUC](https://github.com/ai-se/HPO/issues/12)
